sklearn/metrics/_regression.py

```diff
@@ -137,6 +140,67 @@ def _check_reg_targets(y_true, y_pred, multioutput, dtype="numeric", xp=None):
     return y_type, y_true, y_pred, multioutput
 
 
+def _check_reg_targets_and_floating_dtype(
+    y_true, y_pred, sample_weight, multioutput, xp=None
+):
+    """Ensure that y_true and y_pred correspond to the same regression task.
+
+    This helper automatically selects a floating-point data type with
+    `_find_matching_floating_dtype` when computing with inputs that
+    follow the array API specification.
+
+    To inspect the resulting floating-point data type, users can access the
+    `.dtype` attribute of the returned arrays, e.g., `y_true.dtype` or
+    `y_pred.dtype`.
```

Comment:
@ogrisel, WDYT?

Reply:
scikit-learn users are not expected to call this private function, so let's rephrase as:


```suggestion
def _check_reg_targets_and_floating_dtype(
    y_true, y_pred, sample_weight, multioutput, xp=None
):
    """Ensure that y_true and y_pred correspond to the same regression task.

    This helper automatically selects a floating-point data type with
    `_find_matching_floating_dtype` when computing with inputs that
    follow the array API specification and ensure that all non-None inputs
    are converted to that common dtype if needed before returning them.
```